internal/controller/dnsrecord_controller.go

```diff
@@ -200,13 +204,15 @@ func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, curren
 	var requeueTime time.Duration
 	logger := log.FromContext(ctx)
 
+	logger.V(1).Info("Updating status of DNSRecord")
 	// short loop. We don't publish anything so not changing status
 	if prematurely, requeueIn := recordReceivedPrematurely(current); prematurely {
 		return reconcile.Result{RequeueAfter: requeueIn}, nil
 	}
 
 	// failure
 	if specErr != nil {
+		logger.Error(specErr, "Error reconciling DNS Record")
```

Comment:
IMO we shouldn't need to do this since we should be returning any specErr in the response from this function which would be logged by controller runtime automatically.

Reply:
I was to generous with errors. However, we never return the `specErr` and it could lead to awkward debugging.